oras/provider.py

```diff
@@ -747,7 +748,15 @@ def push(self, *args, **kwargs) -> requests.Response:
 
         # Config is just another layer blob!
         logger.debug(f"Preparing config {conf}")
-        response = self.upload_blob(config_file, container, conf)
+        if config_file is None:
+            with TemporaryDirectory() as tmp:
+                config_file = os.path.join(tmp, "config.json")
+                with open(config_file, "w") as f:
+                    f.write("{}")
+                response = self.upload_blob(config_file, container, conf)
```

Comment:
This response block happens either way, so why not just have the "if none do x" and then remove the else, and no matter what call response = xxxx?

Reply:
If no file is specified, a temporary file must be created before sending, and then the resources provided to create the temporary file must be released.

Therefore, if no file is specified, it is necessary to execute `upload_blob()` in the scope under `with TemporaryDirectory() as tmp:`.

If a single call to `upload_blos` code is desired, the code would be such that it would require execution of `TemporaryDirectory()` even if the file exists.
```python
with TemporaryDirectory() as tmp:
    if config_file is None:
        config_file = os.path.join(tmp, "config.json")
        with open(config_file, "w") as f:
            f.write("{}")
    response = self.upload_blob(config_file, container, conf)
```

Is this the correct modification?

Reply:
I think the "with" context is adding the extra else here - and I see your point about needing to clean up after (and it not working with the context). Would it be too simple to use the already provided functions https://github.com/oras-project/oras-py/blob/02c5bdacb54b5d7d6c152fcdbc0e11cf69b4900f/oras/utils/fileio.py#L168 to create a directory or file, and then just clean it up after? And remove the redundant logic using the with context? :thinking:

Reply:
Fixed using `oras.utils.fileio.get_tmpfile()` .